src/sql/src/query_model/test/catalog.rs

```diff
@@ -0,0 +1,381 @@
+    }
+
+    fn desc(&self, _: &FullObjectName) -> Result<Cow<RelationDesc>, CatalogError> {
+        match &self {
+            TestCatalogItem::BaseTable { desc, .. } => Ok(Cow::Borrowed(desc)),
+            _ => Err(CatalogError::UnknownItem(format!(
+                "{:?} does not have a desc() method",
+                self
+            ))),
+        }
+    }
+
+    fn func(&self) -> Result<&'static Func, CatalogError> {
+        match &self {
+            TestCatalogItem::Func(func) => Ok(func),
+            _ => Err(CatalogError::UnknownFunction(format!(
+                "{:?} does not have a func() method",
+                self
+            ))),
+        }
+    }
+
+    fn source_desc(&self) -> Result<Option<&SourceDesc>, CatalogError> {
+        unimplemented!()
+    }
+
+    fn item_type(&self) -> CatalogItemType {
+        match &self {
+            TestCatalogItem::BaseTable { .. } => CatalogItemType::View,
+            TestCatalogItem::Func(_) => CatalogItemType::Func,
+        }
+    }
+
+    fn create_sql(&self) -> &str {
+        unimplemented!()
+    }
+
+    fn uses(&self) -> &[GlobalId] {
+        unimplemented!()
+    }
+
+    fn used_by(&self) -> &[GlobalId] {
+        unimplemented!()
+    }
+
+    fn subsources(&self) -> Vec<GlobalId> {
+        unimplemented!()
+    }
+
+    fn index_details(&self) -> Option<(&[MirScalarExpr], GlobalId)> {
+        unimplemented!()
+    }
+
+    fn table_details(&self) -> Option<&[Expr<Aug>]> {
+        unimplemented!()
+    }
+
+    fn type_details(&self) -> Option<&CatalogTypeDetails<IdReference>> {
+        unimplemented!()
+    }
+
+    fn connection(&self) -> Result<&Connection, CatalogError> {
+        unimplemented!()
+    }
+}
+
+/// A dummy [`SessionCatalog`] implementation.
+///
+/// This implementation is suitable for use in tests that plan queries which are
+/// not demanding of the catalog, as many methods are unimplemented.
+#[derive(Debug)]
+pub struct TestCatalog {
+    /// Contains the Materialize builtin functions.
+    funcs: HashMap<&'static str, TestCatalogItem>,
+    /// Contains dummy data sources.
+    tables: HashMap<String, TestCatalogItem>,
+    /// Maps (unique id) -> (name of data source).
+    /// Exists to support the `*get_item_by_id` functions.
+    id_to_name: HashMap<GlobalId, String>,
+}
+
+impl Default for TestCatalog {
+    fn default() -> Self {
+        let mut funcs = HashMap::new();
+        for (name, func) in MZ_INTERNAL_BUILTINS.iter() {
+            funcs.insert(*name, TestCatalogItem::Func(func));
+        }
+        for (name, func) in MZ_CATALOG_BUILTINS.iter() {
+            funcs.insert(*name, TestCatalogItem::Func(func));
+        }
+        for (name, func) in PG_CATALOG_BUILTINS.iter() {
+            funcs.insert(*name, TestCatalogItem::Func(func));
+        }
+        Self {
+            funcs,
+            tables: HashMap::new(),
+            id_to_name: HashMap::new(),
+        }
+    }
+}
+
+impl SessionCatalog for TestCatalog {
```

Comment:
We did some work to fully remove the DummyCatalog. Is it worthwhile to add another test impl back? Is the current `Catalog::open_debug_*` stuff still to heavy because it requires a running cockroach?

Reply:
This was just merge skew; removed